ArduCopter/mode_autorotate.cpp

```diff
@@ -0,0 +1,338 @@
+/* -----------------------------------------------------------------------------------------
+    This is currently a SITL only function until the project is complete.
+    To trial this in SITL you will need to use Real Flight 8.
+    Instructions for how to set this up in SITL can be found here:
+    https://discuss.ardupilot.org/t/autonomous-autorotation-gsoc-project-blog/42139
+ -----------------------------------------------------------------------------------------*/
+
+#include "Copter.h"
+#include <AC_Autorotation/AC_Autorotation.h>
+#include "mode.h"
+
+#include <utility>
+
+#if MODE_AUTOROTATE_ENABLED == ENABLED
+
+#define AUTOROTATE_ENTRY_TIME          2.0f    // (s) number of seconds that the entry phase operates for
+#define BAILOUT_MOTOR_RAMP_TIME        1.0f    // (s) time set on bailout ramp up timer for motors - See AC_MotorsHeli_Single
+#define HEAD_SPEED_TARGET_RATIO        1.0f    // Normalised target main rotor head speed (unit: -)
+
+bool ModeAutorotate::init(bool ignore_checks)
+{
+#if FRAME_CONFIG != HELI_FRAME
+    // Only allow trad heli to use autorotation mode
+    return false;
+#endif
+
+    // Check that mode is enabled
+    if (!g2.arot.is_enable()) {
+        gcs().send_text(MAV_SEVERITY_INFO, "Autorotation Mode Change Fail: Mode Not Enabled");
```

Comment:
No big deal, but for the equivalent in Follow mode we send this:
 "gcs().send_text(MAV_SEVERITY_WARNING, "Set FOLL_ENABLE = 1"

Not a blocker, just pointing out a difference.